### DISCUSSION TITLE
Carelink log appearing on screen

### DISCUSSION THREAD

**Discussion by Navid200** [NightscoutFoundation/xDrip#2458] [General]
@benceszasz @tolot27 

Shouldn't this be a log?
Why does it appear on screen?  Is that intentional?
![315474986_10223190013599763_6515614304629376984_n](https://user-images.githubusercontent.com/51497406/201569469-5294caba-d774-411d-b9db-ce4cf4c6be47.jpg)

**Comment by benceszasz** [comment]
@Navid200 yes, this is intentional. During the implementation of CareLink Follower I was trying to follow the base logic of the other already existing similar collectors as templates, like NS follower and Share follower. 
This is the current short status info of the collector, just like for example in case of Nightscout follower or Share follower, where you can also short get similar short error infos. Details can always be found in the log: 
CareLinkFollow : logs of CareLink Follower service
CareLinkFollowDL : logs of CareLink data downloader
CareLinkFollowDP : logs of CareLink data processor

**Reply by Navid200** [reply]
The user says Carelink is working.  The user asks why is there this error on the screen.

Under such circumstances, I didn't think we placed a note on the main screen.  There are even logs that people ask why they see them.  We tell them to ignore the log if the system is working.
But, I cannot tell them to ignore the note on the main screen if the system is working.

**Reply by benceszasz** [reply]
If there is an error message, then there is an error :)
The message appears if an exception is thrown in either CareLink data processing (`CareLinkDataProcessor.processData`) or after that during BGReceive delay calculation (`CareLinkFollowService.updateBgReceiveDelay`). I don't think, that such errors should be hidden, because these exceptions shouldn't happen.
If the user would like to know the cause of the error, then he/she should enable debug logging for CareLinkFollow and CareLinkFollowDP 
Using the debug log I can check and correct the error if it casued by the data processing logic and the CareLink data is correct.
By the way there is a major worldwide system error in CareLink after the CareLink update on 05.11.2022 and Medtronic sent out software outage notifications to the users on 09.11.2022.  As far as I know some are still experiencing errors. Did this error occured in the past 9 days?
